slam-libraries/viam-cartographer/src/slam_service/config.cc

```diff
@@ -69,23 +69,18 @@ void ParseAndValidateConfigParams(int argc, char** argv,
 
     slamService.path_to_data = FLAGS_data_dir + "/data";
     slamService.path_to_map = FLAGS_data_dir + "/map";
-    slamService.configuration_directory = FLAGS_data_dir + "/config/lua_files";
 
     // Find the lua files.
-    path pathToLuasFromConfig(slamService.configuration_directory);
-    if (exists(pathToLuasFromConfig)) {
-        LOG(INFO) << "Using lua files in config folder";
+    auto programLocation = boost::dll::program_location();
+    auto relativePathToLuas = programLocation.parent_path().parent_path();
+    relativePathToLuas.append("share/cartographer/lua_files");
+    if (exists(relativePathToLuas)) {
+        LOG(INFO) << "Using lua files from relative path";
+        slamService.configuration_directory = relativePathToLuas.string();
     } else {
-        auto programLocation = boost::dll::program_location();
-        auto relativePathToLuas = programLocation.parent_path().parent_path();
-        relativePathToLuas.append("share/cartographer/lua_files");
-        if (exists(relativePathToLuas)) {
-            LOG(INFO) << "Using lua files from relative path";
-            slamService.configuration_directory = relativePathToLuas.string();
-        } else {
-            LOG(ERROR) << "No lua files found, looked in "
-                       << pathToLuasFromConfig << " and " << relativePathToLuas;
-        }
+        LOG(ERROR) << "No lua files found, looked in " << relativePathToLuas;
+        LOG(ERROR)
+            << "Use 'make install-lua-carto' to install lua files on system";
```

Comment:
We should throw here, since this should terminate the program.

Reply:
only issue here is currently cartographer tests are intended to run without lua files being present. Since we checking for lua files relative to the binary location, the only way for the tests to locate lua files would be to install the tests to /usr/local/bin before using them, which results in the [tests passing](https://github.com/viamrobotics/slam/actions/runs/3747573312/jobs/6363944760#step:8:11) in CI (as of [this commit](https://github.com/viamrobotics/slam/pull/121/commits/115e6c567bc1e9234791a25105d1ff5708cc4cdd)). 

While this works for CI, it can make the dev experience clunky. To solve this i could possibly add an install target for the cartographer unit tests and add that to `make carto-all`. Should I go ahead and do that & add the throw here? Or is it better to avoid installing more stuff on a user's computer?

also worth noting that if lua files are not found, cartographer does throw because of them at a later step

Reply:
Thanks for explaining. I think it's fine if cartographer throws later.